modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -188,6 +193,13 @@ public void undeployArtifactWithoutUninstall(T artifact) {
     logRequestToUndeployArtifact(artifact);
     try {
       deploymentListener.onUndeploymentStart(artifact.getArtifactName());
+      Optional<Registry> optionalRegistry = ofNullable(artifact.getRegistry());
+
+      optionalRegistry.ifPresent(artifactRegistry -> {
+        Optional<ArtifactStoppedListener> optionalArtifactStoppedListener =
+            artifactRegistry.lookupByName(ARTIFACT_STOPPED_LISTENER);
+        optionalArtifactStoppedListener.ifPresent(artifactStoppedListener -> artifactStoppedListener.mustPersist(false));
```

Comment:
no need to use the intermediate Optional just for this. a simple null check will do.